### PR TITLE
Finish task add gem devise

### PR DESCRIPTION
**Pull request by tqtrung09**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem "turbolinks", "2.3.0"
 gem "jbuilder", "2.2.3"
 gem "sdoc", "0.4.0", group: :doc
 gem "config"
+gem 'devise'
```
Comment (vuducluan) [inline Gemfile]: user ""

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (hellovietnam93) [inline app/models/user.rb]: ident

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<h2>Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```
Comment (hellovietnam93) [inline app/views/devise/sessions/new.html.erb]: remove ()

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<h2>Log in</h2>
```
Comment (hellovietnam93) [inline app/views/devise/sessions/new.html.erb]: I18n

File: app/views/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  <% if devise_mapping.rememberable? -%>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end -%>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
```
Comment (hellovietnam93) [inline app/views/devise/sessions/new.html.erb]: I18n

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,8 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+User.create!(name: "Example User",
```
Comment (hellovietnam93) [inline db/seeds.rb]: remove ()

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,8 @@
...
+             email: "example@railstutorial.org",
```
Comment (hellovietnam93) [inline db/seeds.rb]: ident

File: app/views/devise/sessions/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+  <% if devise_mapping.rememberable? -%>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end -%>
+
+  <div class="actions">
+    <%= f.submit t "login" %>
```
Comment (hellovietnam93) [inline app/views/devise/sessions/new.html.erb]: t("login")

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -5,6 +5,33 @@
...
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              <%= t "views.header.user" %> <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to t("views.header.profile"), "#" %></li>
+              <li><%= link_to t("views.header.setting"), "#" %></li>
+              <li class="divider"></li>
+              <li>
+                <%= link_to t("views.header.signout"), destroy_user_session_path, method: "delete" %>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: method: :delete

File: config/initializers/devise.rb [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` on Rails 4+ applications as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = 'ebc6f949b72e05bf8950ad5d251b22587a962360b0b17ff5b1dcee3f432b3fb9fd34cb76f44fe2cf5b429ce45db9df887b4743a81a581cb87ed51399fc5d2b34'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (hellovietnam93) [inline config/initializers/devise.rb]: ""

**Comment by hellovietnam93** [comment]
OK

**Pull request merged**